### PR TITLE
Assignment 7

### PR DESCRIPTION
**Pull request by ZStuefen**
Completed Assignment but could not check E2E tests because of the net::ERR_CERT_AUTHORITY_INVALID error.

### CODE REVIEW

**Comment by delta-plus** [comment]
I will review this.

**Review by delta-plus — COMMENTED** [review]
Nice work. A few things could be improved but not bad overall.

- Using PlaywrightSharp, automate the following items of the website
  - Verify that when landing on the homepage, Secret Santa is displayed in the navigation banner ✔
  - Verify that all 3 pages (Users, Groups, Gifts) can be reached via clicking on the navigation elements ✔
  - Create a Gift (should assert that the number of gifts after the creation is one greater than the number of gifts before the user was created) ❌ (Test doesn't work. It throws a "waiting for selector" error because the test code has 'await page.TypeAsync("input#UserId", "1");' instead of 'await page.SelectOptionAsync("select#UserId", "1");')
  - Modify the last gift in the list (should assert that the updated data was successfully saved/displayed to the screen) ❌ (Fails with "Assert.AreEqual failed. Expected:<UpdatedGift>. Actual:<UpdatedGiftDrone>." This means you either need to clear the title field before typing into it, use .IsTrue(text.Contains("UpdatedGift")) instead of .AreEqual(), or both.)
  - Delete the last gift in the list (should assert that the number of gifts after the delete is one less than the number of gifts before the deletion) ✔
  - All of these tests should be able to be executed without having to manually start the Web and Api projects ✔
- Using AlpineJS, convert the CRUD pages for Users to be javascript based instead of MVC based
  - Index page should fetch the Users via javascript directly from the API ✔
  - Create page should submit the Users information via javascript directly to the API. Upon successful save, the user should be redirected back to the Users index page ✔
  - Edit page should submit the Users information via javascript directly to the API. Upon successful update, the user should be redirected back to the Users index page ✔
  - Delete opration should submit the request directly to the API via javascript and then refresh the Index page via a javascript call directly to the API ✔

File: SecretSanta/test/SecretSanta.E2E/EndToEnd/EndToEndTests.cs [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+            var gifts = await page.QuerySelectorAllAsync("body > section > section > section");
+            int giftNum = gifts.Count();
+
+            await page.ClickAsync("text=Create");
+
+            await page.TypeAsync("input#Title", "Gift");
+            await page.TypeAsync("input#Url", "google.com");
+            await page.TypeAsync("input#Description", "A new gift");
+            await page.TypeAsync("input#Priority", "1");
+            await page.TypeAsync("input#UserId", "1");
```
Comment (breaman) [inline SecretSanta/test/SecretSanta.E2E/EndToEnd/EndToEndTests.cs:126]: This test failed because you didn't use the SelectOptionAsync

**Review by breaman — COMMENTED** [review]
Overall not bad, but you also committed way too many files which made digging through things a little challenging. You might want to verify that the "Files Changed" seems reasonable (406 was quite a few files).

**Pull request closed without merging**